sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuJsonToStructs.scala

```diff
@@ -36,24 +36,14 @@ case class GpuJsonToStructs(
 
   private def constructEmptyRow(schema: DataType): String = {
     schema match {
-      case struct: StructType if (struct.fields.length > 0) => {
-        val jsonFields: Array[String] = struct.fields.map { field => 
-          field.dataType match {
-            case IntegerType => s""""${field.name}": 0"""
-            case StringType => s""""${field.name}": """""
-            case s: StructType => s""""${field.name}": ${constructEmptyRow(s)}"""
-            case a: ArrayType => s""""${field.name}": ${constructEmptyRow(a)}"""
-            case t => throw new IllegalArgumentException("GpuJsonToStructs currently" +
-              s"does not support input schema with type $t.")
-          }
-        }
-        jsonFields.mkString("{", ", ", "}")
-      }
-      case array: ArrayType => s"[${constructEmptyRow(array.elementType)}]"
-      case _ => "{}"
-    }
+      case struct: StructType if struct.fields.nonEmpty =>
+        s"""{"${escapeFieldName(struct.head.name)}":null}"""
+      case other =>
+        throw new IllegalArgumentException(s"$other is not supported as a top level type")    }
   }
 
+  private def escapeFieldName(fieldName: String): String = fieldName.replace("\"", "\\\"")
```

Comment:
Is [org.apache.commons.text.StringEscapeUtils](https://commons.apache.org/proper/commons-text/javadocs/api-release/org/apache/commons/text/StringEscapeUtils.html) escapeJson/Java of any help?

Reply:
Thanks. I  pushed a commit to use this.